src/Compilers/VisualBasic/Portable/Binding/Binder_Utils.vb

```diff
@@ -1330,9 +1331,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                             Dim binaryOperator = DirectCast(boundExpression, BoundBinaryOperator)
                             ' the right side is expected to be shorter for binary operations, so we use
                             ' recursion for this side.
-                            If GetExpressionConstantValueIfAny(binaryOperator.Right, diagnostics, context) Is Nothing Then
-                                nonConstantDetected = True
-                            End If
+                            GetExpressionConstantValueIfAny(binaryOperator.Right, diagnostics, context)
```

Comment:
Does this change cause some expressions that used to be considered constant to no longer be considered constant? Do we need breaking change documentation for that?

Reply:
> Does this change cause some expressions that used to be considered constant to no longer be considered constant?

Not really. We wouldn't get here if an expression was considered a constant. See the comment right below ```Case BoundKind.BinaryOperator``` and, based on review of the algorithm, I think it is accurate. Given that, unless there is a bug similar to the one I am fixing, an error is going to be reported anyway. What we could end up with before is that, despite an error, a ```ConstantValue``` for a left operand of a binary operator could be returned. That value, clearly, doesn't represent the result of a binary operator itself and perhaps could have wrong type. That probably could affect what our internal structures hold and the constant value that SemanticModel or a symbol reports, but I do not consider that a breaking change. Compiler simply becomes more "accurate" in error scenarios. 

---
In reply to: [540555923](https://github.com/dotnet/roslyn/pull/49912#discussion_r540555923) [](ancestors = 540555923)